src/SonarScanner.MSBuild.Common/Telemetry/TelemetryUtils.cs

```diff
@@ -22,17 +22,78 @@ namespace SonarScanner.MSBuild.Common;
 
 public static class TelemetryUtils
 {
-    private static string[] telemetryProperties = [SonarProperties.ScanAllAnalysis];
-
     public static void AddTelemetry(ILogger logger, AggregatePropertiesProvider aggregatedProperties)
     {
-        foreach (var propertyWithProvider in aggregatedProperties.GetAllPropertiesWithProvider().Where(x => !x.Key.ContainsSensitiveData() && telemetryProperties.Contains(x.Key.Id)))
+        foreach (var kvp in aggregatedProperties.GetAllPropertiesWithProvider().SelectMany(SelectMany))
+        {
+            logger.AddTelemetryMessage(kvp.Key, kvp.Value);
+        }
+    }
+
+    private static IEnumerable<KeyValuePair<string, string>> SelectMany(KeyValuePair<Property, IAnalysisPropertyProvider> argument)
+    {
+        var property = argument.Key;
+        if (property.ContainsSensitiveData()
+            // Further senstive parameters
+            || property.IsKey(SonarProperties.Organization)
+            || property.IsKey(SonarProperties.ProjectKey)
+            // Should be extracted from ServerInfo
+            || property.IsKey(SonarProperties.HostUrl)
+            || property.IsKey(SonarProperties.ApiBaseUrl)
+            || property.IsKey(SonarProperties.SonarcloudUrl)
+            || property.IsKey(SonarProperties.Region))
```

Comment:
I think it would be better to extract the properties in the conditions in arrays to perform a `array.contains`.

will be easier to add/remove parameters and will make the code easier to read IMO.